### DISCUSSION TITLE
Local env when using cargo (sqlx)

### DISCUSSION THREAD

**Discussion by iamquang95** [zed-industries/zed#9264] [Feature Requests]
I'm using sqlx macro in my codebase, which requires `DATABASE_URL` to be defined. How can I add this env to the configuration so the auto compile won't throw errors?

**Comment by elipp** [comment]
You could try:

```
{
  "lsp": {
    "rust-analyzer": {
      "initialization_options": {
        "cargo": {
          "extraEnv": {
                "DATABASE_URL": "postgres://user:pass@127.0.0.1:5432/dbname"
          }
        },
        "server": {
          "extraEnv": {
                "DATABASE_URL": "postgres://user:pass@127.0.0.1:5432/dbname"
          }
        },
        "runnables": {
          "extraEnv": {
                "DATABASE_URL": "postgres://user:pass@127.0.0.1:5432/dbname"
          }
        }
      }
    }
  }
}

```
for me, the only one required is the "cargo" one.

Sorry for necrobump :D

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
